crates/bevy_ecs/src/reflect/component.rs

```diff
@@ -265,6 +266,8 @@ impl ReflectComponent {
 
 impl<C: Component + Reflect + TypePath> FromType<C> for ReflectComponent {
     fn from_type() -> Self {
+        // TODO: Currently we panic of a component is immutable and you use
+        // reflection to mutate it. Perhaps the mutation methods should be fallible?
```

Comment:
I agree with 1 personally. I've also been looking into making certain reflection operations themselves readonly (see #14730) so that should make it easier to extend if such changes would affect `ReflectComponent`.

But yeah feel free to punt this off for now. In the meantime can we add some temporary `# Panics` sections to the docs of the affected methods?

Reply:
Thanks! I've added some `# Panics` sections (or updated existing) to call-out that this will now panic if the component is immutable.